fundamentals/middleware/rate-limit/WebRateLimitAuth/SampleRateLimiterPolicy.cs

```diff
@@ -0,0 +1,33 @@
+using System.Threading.RateLimiting;
+using Microsoft.AspNetCore.RateLimiting;
+
+namespace WebRateLimitAuth;
+
+public class SampleRateLimiterPolicy : IRateLimiterPolicy<string>
+{
+    private Func<OnRejectedContext, CancellationToken, ValueTask>? _onRejected;
+
+    public SampleRateLimiterPolicy(ILogger<SampleRateLimiterPolicy> logger)
+    {
+        _onRejected = (context, token) =>
+        {
+            context.HttpContext.Response.StatusCode = 429;
+            logger.LogWarning($"Request rejected by {nameof(SampleRateLimiterPolicy)}");
```

Comment:
@Kahbazi In the debugger this is hard to hit and when I do, I can't see any logging output. If you run the sample, you get lots of  the following
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at Microsoft.AspNetCore.RateLimiting.RateLimitingMiddleware.CombinedWaitAsync(HttpContext context, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.RateLimiting.RateLimitingMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
The browser seems to refresh the previous successful request and you have no idea there was a problem.

Reply:
@Kahbazi what's the best way to pass in a functioning logger?

Reply:
You could get the logger from `context.HttpContext.RequestService.GetService<ILogger<SampleRateLimiterPolicy>>` in the onRejected method, instead of injecting it in constructor.